core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -468,6 +468,30 @@ trait EntityComponent {
       }
     }
 
+    //noinspection SqlDialectInspection
+    private object CheckForExistingEntityTypeQuery extends RawSqlQuery {
+      val driver: JdbcProfile = EntityComponent.this.driver
+
+      def doesEntityTypeAlreadyExist(workspaceContext: Workspace, entityType: String): ReadAction[Seq[Boolean]] = {
+
+        sql"""select exists (select name from ENTITY
+               where workspace_id=${workspaceContext.workspaceIdAsUUID} and entity_type = $entityType)
+          """.as[Boolean]
+      }
+    }
```

Comment:
this should have an additional `and deleted=0` clause, I think. If the user has (soft-)deleted all entities of a given type, they should be allowed to rename to that type.

Reply:
Added.